datafusion/functions/src/datetime/date_part.rs

```diff
@@ -124,7 +131,7 @@ impl DatePartFunc {
                 ],
                 Volatility::Immutable,
```

Comment:
```suggestion
                Volatility::Stable,
```
The volatility depends on the session timezone.
`current_date` and `now` also use `Stable`